indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,546 @@
+                if (rooms[venue]) {
+                    return $.grep(rooms[venue].data, function(room) {
+                        return room.name.toUpperCase() == query.toUpperCase();
+                    });
+                }
+                return [];
+            }
+
+            function toggleOpenResultsList(inputField) {
+                if (inputField.is(venueInput) && roomInput.closest('.typeahead-container').hasClass('result')) {
+                    roomInput.trigger('typeahead:close-results-list');
+                    venueInput.trigger('typeahead:close-results-list');
+                    venueInput.trigger('typeahead:open-results-list');
+                } else if (inputField.is(roomInput) && venueInput.closest('.typeahead-container').hasClass('result')) {
+                    venueInput.trigger('typeahead:close-results-list');
+                    roomInput.trigger('typeahead:close-results-list');
+                    roomInput.trigger('typeahead:open-results-list');
+                }
+            }
+
+            function resetSelectedOption() {
+                if (latestUsedField.is(venueInput)) {
+                    venueInput.val(latestVenueValue);
+                    if (hiddenData.venue_id) {
+                        highlightOption(venueInput, latestVenueValue)
+                    }
+                } else if (latestUsedField.is(roomInput)) {
+                    roomInput.val(latestRoomValue);
+                    if (hiddenData.room_id) {
+                        highlightOption(roomInput, latestRoomValue)
+                    }
+                }
+            }
+
+            initializeInputs();
+
+            var latestVenueValue = venueInput.val();
+            var latestRoomValue = roomInput.val();
+
+            usingDefault.on('click', function() {
+                toggleDefaultLocation(this.checked);
+            });
+
+            toggleDefaultLocation(usingDefault.prop('checked'), true);
+
+            venueInput.typeahead({
+                source: venues,
+                minLength: 0,
+                searchOnFocus: true,
+                resultContainer: '#' + venueInput.siblings('.typeahead-results').find('.results-list-container').attr('id'),
+                template: {% raw %}'<span id="location-venue-{{id}}">{{name}}</span>'{% endraw %},
+                emptyTemplate: function() { return ''; },  // Used to keep the dropdown list open while there are
+                                                                // no results (required by the 'just-use' option)
+                display: 'name',
+                hint: true,
+                callback: {
+                    onInit: function(node) {
+                        this.query = node.val();  // Updates the results dropdown on init
+                    },
+                    onClickBefore: function() {
+                        validVenue = true;  // Used in the onHideLayout callback to identify the reason of the dropdown
```

Comment:
same here, better to have a longer line than wrapping a single word. there's no hard max line length in js anyway